hyperspy/signal.py

```diff
@@ -4246,6 +4246,88 @@ def integrate_simpson(self, axis):
         s._remove_axis(axis.index_in_axes_manager)
         return s
 
+    def fft(self):
+        """Compute the discrete Fourier Transform.
+
+        This function computes the discrete Fourier Transform over the signal
+        axes by means of the Fast Fourier Transform (FFT) as implemented in
+        numpy.
+
+        Return
+        ------
+        signals.Signal
+
+        Examples
+        --------
+        >>> import scipy.ndimage
+        >>> im = hs.signals.Image(scipy.misc.lena())
+        >>> im.fft()
+        <Image, title: , dimensions: (|512, 512)>
+
+        Notes
+        -----
+        For further information see the documentation of numpy.fft.fftn
+        """
+
+        if self.axes_manager.signal_dimension == 0:
+            raise AttributeError("Signal dimension must be at least one.")
+        ax = self.axes_manager
+        axes = np.arange(ax.signal_dimension) + ax.navigation_dimension
+        im_fft = self._deepcopy_with_new_data(np.fft.fftshift(
+            np.fft.fftn(self.data, axes=axes), axes=axes))
+        shape_fft = self.axes_manager.shape
+        im_fft.metadata.General.title = 'FFT of ' + \
+            im_fft.metadata.General.title
+        for ax, dim in zip(axes, shape_fft):
+            axis = im_fft.axes_manager[ax]
+            axis.scale = 1. / dim / self.axes_manager[ax].scale
+            axis.units = str(self.axes_manager[ax].units) + '$^{-1}$'
+            axis.offset = -axis.high_value / 2.
+        return im_fft
+
+    def ifft(self, shape_ifft=None, axes=None):
+        """
+        Compute the inverse discrete Fourier Transform.
+
+        This function computes the inverse of the discrete
+        Fourier Transform over the signal axes by
+        means of the Fast Fourier Transform (FFT).
+
+        Return
+        ------
+        signals.Signal
+
+        Examples
+        --------
+        >>> import scipy.ndimage
+        >>> im = hs.signals.Image(scipy.misc.lena())
+        >>> imfft = im.fft()
+        >>> imfft.ifft()
+        <Image, title: , dimensions: (|512, 512)>
+
+        Notes
+        -----
+        For further information see the documentation of numpy.fft.ifftn
+
+        """
+
+        if self.axes_manager.signal_dimension == 0:
+            raise AttributeError("Signal dimension must be at least one.")
+        ax = self.axes_manager
+        axes = np.arange(ax.signal_dimension) + ax.navigation_dimension
+        im_ifft = self._deepcopy_with_new_data(np.fft.ifftn(np.fft.ifftshift(
+            self.data, axes=axes), axes=axes).real)
+        im_ifft.metadata.General.title = 'iFFT of ' + \
+            im_ifft.metadata.General.title
+        for ax, dim in zip(axes, self.axes_manager.shape):
+            axis = im_ifft.axes_manager[ax]
+            axis.scale = 1. / dim / self.axes_manager[ax].scale
+            if str(self.axes_manager[ax].units).endswith('$^{-1}$'):
```

Comment:
Why not do something similar with the title? Though I agree that if the data was modified between transforms, having title 'iFFT of FFT of mydata' might be useful, and the units should not "keep history of transforms"